homeassistant/components/webostv/notify.py

```diff
@@ -46,8 +46,8 @@ async def async_send_message(self, message: str = "", **kwargs: Any) -> None:
             if not self._client.is_connected():
                 await self._client.connect()
 
-            data = kwargs.get(ATTR_DATA)
-            icon_path = data.get(CONF_ICON, "") if data else None
+            data = kwargs.get(ATTR_DATA, {})
```

Comment:
`data` may be `None` even when the key is present. It's a bad pattern in the notify integration:
https://github.com/home-assistant/core/blob/1a247f7d1b958b5264e5b40d40483e99fb9bd76c/homeassistant/components/notify/legacy.py#L220

Reply:
I don't understand what this solves looking at the change and the PR description.

Reply:
> `data` may be `None` even when the key is present. It's a bad pattern in the notify integration:
> 
> https://github.com/home-assistant/core/blob/1a247f7d1b958b5264e5b40d40483e99fb9bd76c/homeassistant/components/notify/legacy.py#L220

In this case it would be better to revert back to ` icon_path = data.get(CONF_ICON) if data else None` so that we don't fail both on empty dict and `None`

Reply:
Ok, good. We'll need to correct this fix though to handle the data key having a None value, which is the default.